script.js

```diff
@@ -11,3 +45,471 @@ function hideColumnContents(className) {
+
+        for (const element of columnTwo) {
+          element.style.width = columnTwoWidth + 'px';
+        }
+
+        headerTwo.style.width = columnTwoWidth + 'px';
+      }
+
+      for (const element of columnThree) {
+        element.style.width = columnThreeWidth + 'px';
+      }
+
+      headerThree.style.width = columnThreeWidth + 'px';
+
+      for (const element of columnFour) {
+        element.style.width = columnFourWidth + 'px';
+      }
+
+      headerFour.style.width = columnFourWidth + 'px';
+    }
+  }
+
+  columnlineThree.onmousedown = function(event) {
+    var oldColumnlineThreePosition = columnlineThree.offsetLeft;
+    var columnOneWidth = headerOne.offsetWidth;
+    var columnTwoWidth = headerTwo.offsetWidth;
+    var columnThreeWidth = headerThree.offsetWidth;
+    var columnFourWidth = headerFour.offsetWidth;
+    var oldmousePosition = event.clientX;
+
+    function moveToMouse(windowCoordinatesX) {
+      const newColumnlineLeft = windowCoordinatesX;
+      const newColumnlineLeftIsInMinWidth = newColumnlineLeft >= headerTwo.offsetLeft + minWidth && newColumnlineLeft <= headerThree.offsetLeft + headerThree.offsetWidth - minWidth;
+
+      if (newColumnlineLeftIsInMinWidth) {
+        columnlineThree.style.left = windowCoordinatesX + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+        const diffColumnlineThree = oldColumnlineThreePosition - columnlineThree.offsetLeft;
+
+        let diffIsNegative = diffColumnlineThree < 0;
+
+        if (diffIsNegative) {
+          columnTwoWidth = columnlineThree.offsetLeft - headerOne.offsetWidth;
+          columnThreeWidth = minWidth;
+          for (const element of columnTwo) {
+            element.style.width = columnTwoWidth + 'px';
+          }
+
+          headerTwo.style.width = columnTwoWidth + 'px';
+
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+            
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
+  
+        if (!diffIsNegative) {
+          columnTwoWidth = minWidth;
+          columnThreeWidth = columnThreeWidth + diffColumnlineThree;
+          for (const element of columnTwo) {
+            element.style.width = columnTwoWidth + 'px';
+            
+          }
+
+          headerTwo.style.width = columnTwoWidth + 'px';
+  
+          for (const element of columnThree) {
+            element.style.width = columnThreeWidth + 'px';
+            
+          }
+
+          headerThree.style.width = columnThreeWidth + 'px';
+        }
+        
+        for (const element of columnOne) {
+          element.style.width = columnOneWidth + 'px';
+        }
+
+        headerOne.style.width = columnOneWidth + 'px';
+  
+        for (const element of columnFour) {
+          element.style.width = columnFourWidth + 'px';
+        }
+
+        headerFour.style.width = columnFourWidth + 'px';
```

Comment:
Repetitive code. Introduce a function with the header, column and column width as parameters